Intly 4532 (#33)
aerogear/mobile-walkthrough: walkthroughs/1-ionic-showcase/walkthrough.adoc

* `mosquitto-mqtt-broker` - a message broker

> * `postgresql` - a database

Reviewer: ```suggestion
* `postgresql` - a database
+
Once the pods are running, you can use the GraphQL playground to interact with the server API as described in the next step.
```

> You can use the build-in GraphQL playground to interact with the server API. Navigate to the link:{route-ionic-showcase-server-host}/graphql[GraphQL playground, window="_blank"] and remove any text in the left pane of the screen. Paste the following query/mutation into the left pane of the screen:

Reviewer: ```suggestion
. Navigate to the link:{route-ionic-showcase-server-host}/graphql[GraphQL playground, window="_blank"] and replace the text in the left pane of the screen with the following query and mutation::
```

The showcase app demonstrates the key capabilities provided by the {data-sync-name} framework.
It can be run either as a hybrid mobile application, or a Progressive Web App (PWA).

> . Go to the link:{route-ionic-showcase-server-host}[newly created client app, window="_blank"]

Reviewer: ```suggestion
. Navigate to the link:{route-ionic-showcase-server-host}[Data Sync application user interface, window="_blank"]
```

[subs="attributes"]
----
   window.showcaseConfig = {
     "backend": {
       "serverUrl": "/graphql",
       "wsServerUrl": ((window.location.protocol === "https:") ? "wss://" : "ws://") + window.location.hostname + "/graphql"
     },
     "auth": {
       "realm": "walkthroughs",
    	 "url": " {user-sso-url}/auth",
   	   "ssl-required": "none",
   	   "clientId": "{user-username}-frontend",
   	   "public-client": true,
   	   "confidential-port": 0
     }
   };
----
****

[type=verificationFail]
****
Verify that you followed each step in the procedure above.  If you are still having issues, contact your administrator.
****
[time=10]

=== Configure and protect the backend app

. Login to the realm in the link:{sso-realm-url}[{customer-sso-name}, window="_blank"].
. Create a new client for the backend:
.. Choose *Clients* under *Configure* in the menu.
.. Click on the *Create* button on the top-right corner to start creating a new client.
.. Enter `{user-username}-server` for Client ID.  Do not change the rest of the fields.
.. Save the client. You should see the *Settings* page of the client.
.. Change the `Access Type` to `bearer-only`

> .. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Use the `Download` button to save the configuration file to a directory on the computer.

Reviewer: ```suggestion
.. Click on the *Installation* tab, and select `Keycloak OIDC JSON` format. Use the `Download` button to save the `keycloak.json` configuration file.
```

=== Configure and protect the frontend app

> . Make sure you have access to a realm in the link:{sso-realm-url}[{customer-sso-name}, window="_blank"]. If you do not, you can ask your customer-admin user to create one for you. 

Reviewer: this seems wrong?

Author: Still a draft PR, not ready for review yet.

Reviewer: don't think we need this step? it's like 'make sure RH haven't messed up'

----
keycloak.json
----

> .. For *Value*, click *Browse* and load the json file that you downloaded previously.

Reviewer: switched from first order list to second order for no good reason?

To explore data sync features, you should run multiple instances of the showcase app using different browsers.
For example, use the browser on your mobile device as well as using the browser on your laptop.

To get the url of your app:

> . Navigate to the {openshift-host}/console/project/{walkthrough-namespace}/browse/routes[Routes page of the OpenShift console]

Reviewer: can replace all this with 

Navigate to the link:{route-ionic-showcase-server-host}[Data Sync user interface]
